xwiki-platform-tools/xwiki-platform-tool-packager-plugin/src/main/java/com/xpn/xwiki/tool/backup/DataMojo.java

```diff
@@ -71,4 +75,40 @@ public void executeInternal() throws MojoExecutionException
 
         this.extensionHelper.install(this.includes, installRequest, "wiki:xwiki", null);
     }
+
+    @Override
+    protected void after() throws MojoExecutionException
+    {
+        consumeIndexingQueue();
+        super.after();
+    }
+
+    /**
+     * Waits for the document indexing tasks to be completed before continuing.
+     *
+     * @throws MojoExecutionException in case of error when resolving the components or when stopping the thread
+     */
+    private void consumeIndexingQueue() throws MojoExecutionException
+    {
+        ComponentManager componentManager =
+            (ComponentManager) this.oldCoreHelper.getXWikiContext().get(ComponentManager.class.getName());
+        try {
+            // Manually start the consumer threads as it is not automatically started in the packaging setup.
+            // This is also convenient as we can be sure that the tasks are all queued before being consumed.
+            TaskManager taskManager = componentManager.getInstance(TaskManager.class);
+            if (taskManager instanceof DefaultTasksManager) {
+                ((DefaultTasksManager) taskManager).startThread();
+            }
+
+            getLog().info("Waiting for the document indexing queue to be empty...");
```

Comment:
Maybe indicate how big is the queue before we start to wait for it to empty (to give a clue of how long it might take).